help_to_heat/frontdoor/interface.py

```diff
@@ -534,53 +534,19 @@ def get_epc_scotland(self, uprn):
 
         return {"uprn": epc.uprn, "rating": epc.rating, "date": epc.date}
 
-    def get_address_and_epc_rrn(self, building_name_or_number, postcode):
-        data = self.__try_with_authentication(lambda api: api.get_address_and_rrn(building_name_or_number, postcode))
-        address_and_epc_details = data["data"]["assessments"]
+    def get_address_and_epc_lmk(self, building_name_or_number, postcode):
+        data = self.__try_connection(lambda api: api.get_address_and_lmk(building_name_or_number, postcode))
+        address_and_epc_details = data["rows"]
         return address_and_epc_details
 
-    def get_epc_details(self, rrn):
-        return self.__try_with_authentication(lambda api: api.get_epc_details(rrn))
+    def get_epc_details(self, lmk):
+        return self.__try_connection(lambda api: api.get_epc_details(lmk))
 
-    def __try_with_authentication(self, callback):
+    def __try_connection(self, callback):
         try:
-            token = self.__get_access_token()
-            return callback(EPCApi(token))
+            return callback(EPCApi())
         except requests.exceptions.RequestException as e:
-            status_code = e.response.status_code
-            if status_code == HTTPStatus.UNAUTHORIZED:
-                new_token = self.__update_access_token()
-                return callback(EPCApi(new_token))
-            else:
-                raise e
-
-    def __get_access_token(self):
-        token = self.__retrieve_token()
-        if token is not None:
-            return token
-        else:
-            token = self.__update_access_token()
-            return token
-
-    def __update_access_token(self):
-        access_token = EPCApi(None).get_access_token()
-        self.__save_token(access_token)
-        return access_token
-
-    def __retrieve_token(self):
-        token = models.AccessToken.objects.first()
-        if token is not None:
-            return token.access_token
-        else:
-            return None
-
-    def __save_token(self, new_token):
-        saved_token = models.AccessToken.objects.first()
-        if saved_token is not None:
-            saved_token.access_token = new_token
-        else:
-            saved_token = models.AccessToken(access_token=new_token)
-        saved_token.save()
+            raise e
```

Comment:
Is there any benefit to catching if we're going to re-throw straight away? (and then is there any benefit to this wrapper method at all?)

Reply:
perhaps the wrapper can stay as it means it's only needed to construct `EPCApi` in one place, perhaps could be renamed

agree the try isn't much needed

Reply:
left the wrapper method for now

Reply:
Not totally confident in my understanding of patching in Python, but I don't immediately see why the tests wouldn't wok with e.g.
```
api = EpcApi()
data = api.get_address_and_lmk(building_name_or_number, postcode)
```
? It should be equivalent. Would be quite a lot clearer than what we have at the moment - feels like machinery for the sake of it.